lib/opam_build.ml

```diff
@@ -51,18 +58,27 @@ let rec get_root_opam_packages = function
 let download_cache = "--mount=type=cache,target=/home/opam/.opam/download-cache,uid=1000"
 
 let install_project_deps ~base ~opam_files ~selection ~for_user =
-  let { Selection.packages; commit; variant } = selection in
+  let { Selection.packages; post_packages; commit; variant; pin_depends } = selection in
   let groups = group_opam_files opam_files in
   let root_pkgs = get_root_opam_packages groups in
   let non_root_pkgs = packages |> List.filter (fun pkg -> not (List.mem pkg root_pkgs)) in
   let download_cache_prefix = if for_user then "" else download_cache ^ " " in
+  (* Remove root packages from post packages. This can happen when a package
+     has a 'post' dependency on an other root package. *)
+  let post_packages = List.filter (fun p -> not (List.mem p root_pkgs)) post_packages in
```

Comment:
This case happens on Irmin, `irmin` requires `irmin-mem.dev` but `irmin-mem.dev` was pinned to an empty source tree, Opam seems to do nothing. We need to filter them because now this case will be an error.